datalab/datalab_session/util.py

```diff
@@ -97,6 +98,42 @@ def get_archive_from_basename(basename: str) -> dict:
 
   return results
 
+def get_hdus(basenames) -> list[fits.HDUList]:
+  """
+  Returns a list of HDULists for the given basenames
+  Warning: this function returns an opened file that must be closed after use
+  """
+  if isinstance(basenames, str):
+    basenames = [basenames]
+
+  hdu_list = []
+
+  # use the basename to fetch and create a list of hdu objects
+  for basename in basenames:
+    basename = basename.replace('-large', '').replace('-small', '')
+
+    if cache.get(basename) is not None:
+      hdu_list.append(cache.get(basename))
+      continue
+
+    archive_record = get_archive_from_basename(basename)
+
+    try:
+      fits_url = archive_record[0].get('url', 'No URL found')
+    except IndexError:
+      RuntimeWarning(f"No image found with specified basename: {basename}")
+      continue
+
+    hdu = fits.open(fits_url, use_fsspec=True)
+    cache.set(basename, hdu)
+
+    hdu_list.append(hdu)
+
+  if len(hdu_list) == 1:
+    return hdu_list[0]
+  else:
+    return hdu_list
```

Comment:
Maybe we can refactor this function to just get a single HDU by key, and put that key in as a function parameter, rather than having to do all of this logic to build up an HDUList.

We can default the extension_name arg to SCI if we want, so the function signature can look like

```
get_hdu(basename, extension: str = 'SCI'):
    ...
```

Reply:
anything that makes a function more general makes me happy